projects/driver_controls/src/event_arbiter.c

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+
+#include "event_arbiter.h"
+#include "input_event.h"
+#include "log.h"
+
+static FSM *s_driver_fsms[EVENT_ARBITER_MAX_FSMS];
+static EventArbiterCheck s_event_checks[EVENT_ARBITER_MAX_FSMS];
+
+static uint8_t s_num_active_fsms = 0;
+
+static bool prv_event_permitted(Event *e) {
+  bool permitted;
+
+  for (uint8_t i = 0; i < s_num_active_fsms; i++) {
+    permitted = s_event_checks[i](e);
+    if (!permitted) {
+      return false;
+    }
+  }
+  return true;
+}
+
+StatusCode event_arbiter_init() {
+  for (uint8_t i = 0; i < EVENT_ARBITER_MAX_FSMS; i++) {
+    s_driver_fsms[i] = NULL;
+    s_event_checks[i] = NULL;
+  }
+  s_num_active_fsms = 0;
+}
+
+StatusCode event_arbiter_add_fsm(FSM *fsm, EventArbiterCheck default_checker) {
+  if (s_num_active_fsms == EVENT_ARBITER_MAX_FSMS) {
+    return STATUS_CODE_RESOURCE_EXHAUSTED;
```

Comment:
You need to wrap non-OK status codes in `status_code` or `status_msg` so that the global status gets updated.